Adds gorillas and cargorillas

S34NW opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do
Adds gorillas and cargorillas

Gorilla:
- 220 HP
- 15-18 damage to mobs
- 20 damage to objects
- Can smash walls
- 0.5x speed
- 1.5x Burn, Toxin, Oxy damage
- Immune to clone and stamina damage
- Can carry one crate
- Can stand up on two legs if they want to
- Throws mobs they punch
- Can dismember unconscious mobs
- 1% chance of mutating from monkeys with over 1500 rads (balance figures subject to change)
- Hostile (they will find you and they will kill you)
- 1% chance of mutating when any carbon's `randmutb()` is called (cloning, genetics DNA modifiers, rad storms, and rarely by unstable mutagen)

Rampaging Gorilla:
- Red and runs at normal speeds
- Has 350 health
- Does more damage.
- Takes 1.25x Brute damage (Unlike the 1x parent)
- Takes 1x Burn and Oxy damage (Unlike the 1.5x parent)

Cargorilla:
- Can pick up 2 crates at the same time
- Has cargo access
- Pacifist
- Will eventually spawn in cargo in a subsequent PR

Added cargorilla crates, 250 credits. (mulebot crate cost)
Added gorilla crates, 1000 credits.

Added Magillitis bio-chip, turning you into rampaging gorilla when you use the action.

Added Magillitis bio-chip implanter to RD and Genetics traitor uplinks, 5TC.

Removed random crappy un-used big crates.

Added a bunch of mobs to the "jungle" faction so the gorilla will leave them alone, went with ones that matched thematically.

Moved simplemob var comments to autodoc, because that was annoying as hell.
<!-- Include a small to medium description of what your PR changes. -->
<!-- Document all changes, as not doing this may delay reviews or even discourage maintainers from merging your PR! -->
<!-- If your PR fixes an issue, add "Fixes #1234" somewhere in the PR description. This will automatically close the bug upon PR submission. -->

## Why It's Good For The Game
Rampaging gorillas?

https://user-images.githubusercontent.com/12197162/230654007-a507e232-c74f-49f9-82e7-df9446ea61a2.mp4


<!-- Add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Images of changes
![image](https://user-images.githubusercontent.com/12197162/230438076-103d307e-632a-4a25-9cc7-29c02a3d20c3.png)
<!-- If you did not make a map or sprite edit, you may delete this section. You may include a gif or mp4 of your feature if you want. -->

## Testing
Gorilla crate picking, punching things, etc.
<!-- How did you test the PR, if at all? -->

## Changelog
:cl:
add: Added gorillas and cargorillas
add: Added Magillitis bio-chip implanter to RD and Genetics traitor uplinks, 5TC
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->
<!-- If a PR has no impact on players (i.e. a code refactor that does not change functionality) then the entire Changelog heading and contents can be removed. -->

S34NW commented:
```
FAIL /datum/unit_test/emote 0.1s
	Emote /datum/emote/living/simple_animal/gorilla/ooga is usable, but not assigned a keybind.
```

guh

Probablyjay19 commented:
![image](https://user-images.githubusercontent.com/87214315/230453593-3e9d925a-3168-4712-bdc9-e79aa4319e2f.jpeg)

SynthTwo commented:
gorillas :) gorilla friend

Leanfrog commented:
https://www.youtube.com/shorts/yW8V--TrlJM

S34NW commented:
> ![image](https://user-images.githubusercontent.com/87214315/230453593-3e9d925a-3168-4712-bdc9-e79aa4319e2f.jpeg)

Added

Probablyjay19 commented:
> > ![image](https://user-images.githubusercontent.com/87214315/230453593-3e9d925a-3168-4712-bdc9-e79aa4319e2f.jpeg)
> 
> 
> 
> Added

![image](https://user-images.githubusercontent.com/87214315/230494997-6fdcbd31-8050-4703-9048-97bc2af4db03.jpeg)

(Although, 5 TC is dirt fucking cheap. On TG it's 15 TC 🥶)

S34NW commented:
> 
> (Although, 5 TC is dirt fucking cheap. On TG it's 15 TC 🥶)

True, but genetics is shit when it comes to traitor opportunities. And once you go gorilla, you can never go back.

Denthamos commented:
> > (Although, 5 TC is dirt fucking cheap. On TG it's 15 TC 🥶)
> 
> True, but genetics is shit when it comes to traitor opportunities. And once you go gorilla, you can never go back.

There needs to be a reference when a geneticist turns themself into a gorilla, to the meme "Return to monke"

S34NW commented:
```
[2023-04-07T11:11:02] Runtime in dna2_helpers.dm,244: Attempting to reset DNA from a missing head!
  proc name: head traits to dna (/datum/dna/proc/head_traits_to_dna)
  src: /datum/dna (/datum/dna)
  call stack:
  /datum/dna (/datum/dna): head_traits_to_dna
  /datum/dna (/datum/dna): ResetUIFrom
  /datum/dna (/datum/dna): ready_dna
  the monkey (872) (/mob/living/carbon/human/monkey): setup_dna
  the monkey (872) (/mob/living/carbon/human/monkey): Initialize
  the monkey (872) (/mob/living/carbon/human/monkey): Initialize
  Atoms (/datum/controller/subsystem/atoms): InitAtom
  Atoms (/datum/controller/subsystem/atoms): InitializeAtoms
  Atoms (/datum/controller/subsystem/atoms): Initialize
  Atoms (/datum/controller/subsystem/atoms): call_init
  Master (/datum/controller/master): Initialize```

Wasn't me, I swear

Sirryan2002 commented on code/modules/mob/living/carbon/human/species/monkey.dm:
```diff
@@ -38,6 +39,12 @@
+	. = ..()
+	if(H.radiation > RAD_MOB_GORILLIZE && prob(RAD_MOB_GORILLIZE_PROB))
+		H.gorillize()
+		return
```
```suggestion
```
dangling return

Sirryan2002 commented on code/modules/mob/living/simple_animal/hostile/gorilla/gorilla.dm:
```diff
@@ -0,0 +1,243 @@
+		to_chat(domesticated, "<span class='warning'>You can't get on all fours while carrying something!</span>")
+		return
+
+	gorilla.is_bipedal =! gorilla.is_bipedal // Toggle
```
```suggestion
	gorilla.is_bipedal = !gorilla.is_bipedal // Toggle
```
just to make this more readable

Sirryan2002 commented on code/modules/mob/living/simple_animal/hostile/gorilla/gorilla.dm:
```diff
@@ -0,0 +1,243 @@
+		return
+
+	gorilla.is_bipedal =! gorilla.is_bipedal // Toggle
+	gorilla.visible_message("<span class='notice'>[gorilla] [gorilla.is_bipedal ? "stands up menacingly." : "drops back to all fours."]</span>", "<span class='notice'>You [gorilla.is_bipedal ? "stand up" : "get down on all fours."]</span>", "<span class='notice'>You hear the sound of a gorilla rustling.</span>")
```
```suggestion
	gorilla.visible_message("<span class='notice'>[gorilla] [gorilla.is_bipedal ? "stands up menacingly." : "drops back to all fours."]</span>",
		"<span class='notice'>You [gorilla.is_bipedal ? "stand up" : "get down on all fours."]</span>", 
		"<span class='notice'>You hear the sound of a gorilla rustling.</span>")
```
multiline for readability

Sirryan2002 commented on code/modules/mob/living/simple_animal/hostile/gorilla/gorilla.dm:
```diff
@@ -0,0 +1,243 @@
+	gold_core_spawnable = NO_SPAWN
+	a_intent = INTENT_HELP
+	unique_pet = TRUE
+	/// The ID card that the gorilla is currently wearing.
+	var/obj/item/card/id/access_card
```
could you move this below the inherited vars please

Henri215 commented on code/modules/mob/living/simple_animal/hostile/gorilla/gorilla.dm:
```diff
@@ -0,0 +1,246 @@
+/mob/living/simple_animal/hostile/gorilla
+	name = "Gorilla"
+	desc = "A ground-dwelling, predominantly herbivorous ape that inhabits the forests of central Africa."
```
What do you think about saying that they are from earth instead of central africa? Sounds too specific since only humans are from this planet.

S34NW replied:
Sure

Henri215 commented on code/modules/mob/living/simple_animal/hostile/gorilla/gorilla.dm:
```diff
@@ -0,0 +1,246 @@
+/mob/living/simple_animal/hostile/gorilla
+	name = "Gorilla"
+	desc = "A ground-dwelling, predominantly herbivorous ape that inhabits the forests of central Africa on Earrh."
```
```suggestion
	desc = "A ground-dwelling, predominantly herbivorous ape that inhabits the forests of central Africa on Earth."
```

lewcc replied:
```suggestion
	desc = "A ground-dwelling, predominantly herbivorous ape that inhabits the forests of central Africa on Earth."
```

lewcc commented on code/modules/reagents/chemistry/reagents/misc_reagents.dm:
```diff
@@ -732,3 +698,17 @@
+	..()
+	if((istype(M)) && current_cycle >= 10)
+		M.gorillize(TRUE)
+		to_chat(M, "<span class='notice'>You feel your muscles swell and your hair grow as you return to monke.</span>")
```
This should probably be a pretty noticeable message, maybe even more noticeable
```suggestion
		to_chat(M, "<span class='userdanger'>You feel your muscles swell and your hair grow as you return to monke.</span>")
```

Samman166 commented:
Are the cargorillas player controlled? Or just NPCs

S34NW commented:
> How does the cargoilla sprite look with the wooden crate type? the ones the tanks, floodlights and live animals use? Im slightly worried that it might look janky.

The big crate can not be carried by gorillas

> Are the cargorillas player controlled? Or just NPCs

They can be player controlled, in the same way that Betsy, Clucky, etc. can be

ATP-Engineer commented:
I like the idea of 5 TC serum having opportunity as a last resort to a regular loadout. OR... Two serums and mindslave implant.

SynthTwo commented:
God I am so excited for gorillas

Sirryan2002 commented:
waiting on TM for additional reviews / approval

GDNgit commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -188,3 +188,10 @@
+/obj/item/reagent_containers/hypospray/autoinjector/magillitis
+	name = "experimental autoinjector"
+	desc = "A custom-frame needle injector with a small single-use reservoir, containing an experimental serum. Unlike the more common medipen frame, it cannot pierce through protective armor or space suits, nor can the chemical inside be extracted."
+	icon_state = "gorillapen"
```
this doesn't exist in your current branch, causing the injector to be invisible in game

S34NW replied:
Thanks, i deleted this to fix the issue.

Zorazi-Teo-Daedalus commented:
Quick question. Is it intended that you're unable to put pet collars on gorillas? Or at least the Cargorilla. I tried because one went and got itself an ID, so I was gonna affix it to a pet collar and put it on them but was unable to attempt to put the collar on them like you can with some other mobs.

S34NW commented:
> Quick question. Is it intended that you're unable to put pet collars on gorillas? Or at least the Cargorilla. I tried because one went and got itself an ID, so I was gonna affix it to a pet collar and put it on them but was unable to attempt to put the collar on them like you can with some other mobs.

Not in my current plans, but someone could add it after. I don't want to further delay the PR.
I fixed access though, cargorillas should have cargo access.

S34NW commented:
Latest commit:
Fixed cargorilla access.
Removed magillitis reagent and made it a bio-chip to prevent people murdering their targets via gorilla juice injector.
Buffed rampaging gorilla.
Swapped paralyse out for 40 stam loss on gorilla hit.

Venuska1117 commented:
Cargorillas cannot have name changed when they got equip pet collar (they have slot for it)

And people want be able to pick up people under hand and carry them around like crates (rotated 90 human sprite here)
and... some wanted be able to pick coffin and be coffin taxi gorilla (with killed me inside)

SteelSlayer commented:
I have a report from somebody on discord that the radiation storm event turned _dead_ bodies in the morgue into gorillas. Have not confirmed this myself however.

CinnamonSnowball commented:
> I have a report from somebody on discord that the radiation storm event turned _dead_ bodies in the morgue into gorillas. Have not confirmed this myself however.

I'm fairly sure I've seen it before, it's more of a problem with bodies in morgue trays being affected by radiation in general

Coolrune206 commented:
After seeing these TM'd on live for a bit, I can't say I'm a fan of the frequency of Gorillas appearing. Radiation storms turn people into Gorillas too frequently, and Genetics seems to be turning themselves into Gorillas somehow every fuckin' round.

I feel Gorillas should be significantly rarer, and crewmembers probably should never be turned into them at all, else I don't think this will be particularly beneficial to Paradise.

Sirryan2002 commented on sound/creatures/gorilla.ogg:
this is really loud in-game

S34NW replied:
lowered

S34NW commented:
> After seeing these TM'd on live for a bit, I can't say I'm a fan of the frequency of Gorillas appearing. Radiation storms turn people into Gorillas too frequently, and Genetics seems to be turning themselves into Gorillas somehow every fuckin' round.
> 
> I feel Gorillas should be significantly rarer, and crewmembers probably should never be turned into them at all, else I don't think this will be particularly beneficial to Paradise.

>I have a report from somebody on discord that the radiation storm event turned dead bodies in the morgue into gorillas. Have not confirmed this myself however.

I dropped the prob from 1% to 0.1%, hopefully this will feel more organic. I've also stopped dead mobs from being gorrila'd

CinnamonSnowball commented on code/modules/mob/living/simple_animal/hostile/gorilla/gorilla.dm:
```diff
@@ -0,0 +1,258 @@
+/mob/living/simple_animal/hostile/gorilla
+	name = "Gorilla"
```
```suggestion
	name = "gorilla"
```

CinnamonSnowball commented on code/modules/mob/living/simple_animal/hostile/gorilla/gorilla.dm:
```diff
@@ -0,0 +1,258 @@
+	update_icon(UPDATE_OVERLAYS | UPDATE_ICON_STATE)
+
+/mob/living/simple_animal/hostile/gorilla/cargo_domestic
+	name = "Cargorilla" // Overriden, normally
```
```suggestion
	name = "cargorilla" // Overriden, normally
```

CinnamonSnowball reviewed (commented):
Capitalized names!! Gnarg!!

tipoima commented:
Does the chip bundle get the new chip?
